src/coreclr/src/vm/frames.cpp

```diff
@@ -1657,12 +1657,6 @@ void HelperMethodFrame::Push()
 
     // Push the frame
     Frame::Push(pThread);
-
-    if (!pThread->HasThreadStateOpportunistic(Thread::TS_AbortRequested))
```

Comment:
Can removing this fast path cause perf regressions?

Reply:
It may be better to just move `INSTALL_MANAGED_EXCEPTION_DISPATCHER` into `HELPER_METHOD_FRAME_BEGIN_EX_BODY` right after `__helperframe.Push();`.

Reply:
@jkotas I've updated it along the lines you've suggested. I have originally wanted to do it this way, but I got worried about moving the actual frame popping inside of the try regions created by the INSTALL_MANAGED_EXCEPTION_DISPATCHER / UNINSTALL_MANAGED_EXCEPTION_DISPATCHER  and INSTALL_UNWIND_AND_CONTINUE_HANDLER / UNINSTALL_UNWIND_AND_CONTINUE_HANDLER.

But after thinking about it, it seems to be fine as it is the last thing that happens in the try region.